conn.go

```diff
@@ -223,6 +223,30 @@ func isValidReceivedCloseCode(code int) bool {
 	return validReceivedCloseCodes[code] || (code >= 3000 && code <= 4999)
 }
 
+// BufferPool represents a pool of buffers.
+type BufferPool interface {
+	// Get gets a buffer from the pool. Get must not return nil.
+	Get() []byte
+	// Put returns a buffer to the pool.
+	Put(buf []byte)
+}
+
+type syncBufferPool struct{ pool sync.Pool }
+
+func (p *syncBufferPool) Get() []byte {
+	return p.pool.Get().([]byte)
+}
+
+func (p *syncBufferPool) Put(buf []byte) {
```

Comment:
Note that the buffers in this pool will grow over time: unless they are resized by the user, the pool will inevitably contain only the largest buffers used by your program, even if most of your buffers were small.

This can lead to growing memory usage between GC cycles, and increased GC pressure. 

https://github.com/golang/go/issues/23199 covers this well.

Reply:
The byte slices in this scenario all have the same capacity.